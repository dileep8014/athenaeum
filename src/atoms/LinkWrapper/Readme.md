LinkWrapper Examples:

    <LinkWrapper href='#'>This is a link!</LinkWrapper>
