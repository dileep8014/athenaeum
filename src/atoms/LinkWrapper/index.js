import React from 'react';
import PropTypes from 'prop-types';
import styles from './link_wrapper.module.scss';
import classnames from 'classnames';
import _ from 'lodash';

function LinkWrapper ( props ) {
  const {
    className,
    children,
  } = props;

  return (
    <a
      { ..._.omit(props, [ 'children', 'className' ]) }
      className={ classnames(styles.link, className) }>
      { children }
    </a>
  )
}

LinkWrapper.propTypes = {
  /**
   * This wrapper will take any props that you would use
   * for an anchor `<a>` tag. `href` is just an example.
   */
  href: PropTypes.string
}

export default LinkWrapper;
