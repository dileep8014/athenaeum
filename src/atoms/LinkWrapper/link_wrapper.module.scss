.link {
  color: color('neutral-4');
  display: inline-block;
  text-decoration: underline;
}

@each $key, $value in $colors-map {
  .#{$key} { color: $value; }
}
