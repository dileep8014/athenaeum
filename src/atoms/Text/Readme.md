Text Example:

```jsx

    <div>
      <Text type={0}>Type 0</Text>
      <Text type={1}>Type 1</Text>
      <Text type={2}>Type 2</Text>
      <Text type={3}>Type 3</Text>
      <Text type={4}>Type 4</Text>
      <Text type={5}>Type 5</Text>
      <Text type={6}>Type 6</Text>
      <Text type={7}>Type 7</Text>
    </div>
```
