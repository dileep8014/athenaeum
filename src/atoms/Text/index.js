import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import omit from 'lodash/omit';

import colors from 'atoms/Color/colors.scss';
import styles from './text.module.scss';

const getTag = (props) => {
  const {
    tag,
    semibold,
    variant,
  } = props;

  if (semibold) return 'strong';
  if (variant === 'italics') return 'em';
  if (variant === 'strikethrough') return 'span';

  return tag;
};

const getFont = ({
  size, type, font, a
}) => {
  const fontSize = type || size;

  // TODO: Remove 'a' prop as an option from component
  if (a || font === 'a') {
    return styles[`type-a-${fontSize}-bold`];
  } if (font === 'c') {
    return styles['type-c-7-regular'];
  }

  return styles[`type-b-${fontSize}-medium`];
};

const convertChild = (child) => {
  // Check children for other TextComponents
  // and automatically convert them into <spans> if they are <p>.
  if (!child) { return null; }

  if (child.type === Text && child.props.tag === 'p') {
    return React.cloneElement(child, { tag: 'span' });
  }

  return child;
};

const getWeight = ({ font, bold, semibold }) => {
  if (font !== 'b') { return false; }
  if (bold || semibold ) { return styles.bold; }

  return false;
};

function Text(props) {
  // TODO: Consider adding an html sanitizer

  const {
    align,
    children,
    className,
    color,
    italic,
    spaced,
    variant,
    inherit,
    inheritSize,
    inheritColor,
    dangerouslySetInnerHTML,
    ...rest
  } = props;

  if (!children && !dangerouslySetInnerHTML) return null;

  const font = getFont(props);
  const weight = getWeight(props);
  const tag = getTag(props);

  const classes = [
    weight && weight,
    color && colors[color],
    variant && styles[variant],
    align && styles[align],
    spaced && styles['spaced'],
    italic && styles['italic'],
    className && className,
    font && font,
    inherit && styles['inherit'],
    inheritSize && styles['inherit-size'],
    inheritColor && styles['inherit-color']
  ];

  return React.createElement(
    tag,
    {
      className: classnames(classes),
      ...omit(
        rest,
        [ 'font', 'semibold', 'size', 'tag', 'type', 'bold' ]
      )
    },
    React.Children.map(children, convertChild)
  );
}

Text.propTypes = {

  /*
   * If neither children nor dangerouslySetInnerHTML is set, this component will return null
   */
  dangerouslySetInnerHTML: PropTypes.shape({
    __html: PropTypes.string
  }),

  /*
   * Text alignment
   */
  align: PropTypes.oneOf([
    'left',
    'right',
    'center'
  ]),

  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * You can use any html element text tag
   */
  tag: PropTypes.string,

  /**
   * Determines typography class
   *
   * Types for `a`: `1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11`
   *
   * Types for `b`: `5, 6, 7, 8, 10, 12`
   *
   * Types fo `c`: `1`
   */
  size: PropTypes.number,

  /**
   * Text decoration
   */
  variant: PropTypes.oneOf([
    'strikethrough',
    'underline',
    'fineprint',
    'label'
  ]),

  /**
   * Color style, see [Colors](#colors) for appropriate values
   */
  color: PropTypes.string,

  /**
   * Possible font types are `a`, `b`, and `c`
   */
  font: PropTypes.string,

  /**
   * Adds letter spacing. Use with `A9` and `A11` font
   */
  spaced: PropTypes.bool,

  /**
   * Makes text bold
   */
  bold: PropTypes.bool,

  /**
   * Adds italic
   */
  italic: PropTypes.bool,

  /**
   * Deprecated. Use 'size' instead
   */
  type: PropTypes.number,

  /**
   * Deprecated
   */
  weight: PropTypes.oneOf([]),

  /**
   * Deprecated
   */
  light: PropTypes.bool,

  /**
   * Deprecated
   */
  semibold: PropTypes.bool,

  /**
   * Deprecated
   */
  a: PropTypes.bool,

  /**
   * Deprecated
   */
  b: PropTypes.bool,

  /**
   * Inherit font-size, line-height, and color from parent
   */
  inherit: PropTypes.bool,

  /**
   * Inherit font-size from parent
   */
  inheritSize: PropTypes.bool,

  /**
   * Inherit color from parent
   */
  inheritColor: PropTypes.bool,
};

Text.defaultProps = {
  color: 'primary-3',
  font: 'b',
  inherit: true,
  size: 8,
  tag: 'p',
};

export default Text;
