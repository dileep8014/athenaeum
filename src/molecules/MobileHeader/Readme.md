### MobileHeader Example:

```jsx
    <MobileHeader
      text={
        <LinkWrapper
          type='secondary'
          onClick={() => alert('save progress')}
        >
          <Text
            size={10}
            type='b'
          >
            Save your progress
          </Text>
        </LinkWrapper>
      }
      quote={
        <Text
          size={6}
          font='a'
        >
          <Text tag='sup' size={10} font='a'>$</Text>39.50 <Text tag='span' size={10} font='a'>/mo</Text>
        </Text>
      }
      dropdown={{
        menuActive: false,
        onClick: () => alert('dropdown'),
      }}
    />
```
