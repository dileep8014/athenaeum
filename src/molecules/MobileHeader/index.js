import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'atoms/Icon';
import styles from './mobile_header.module.scss';

function MobileHeader(props) {
  const {
    className,
    text,
    quote,
    dropdown,
  } = props;

  const classes = [
    styles['mobile-header'],
    className,
  ];

  return (
    <div className={classnames(...classes)}>
      <div className={styles['main-text']}>
        {text}
      </div>

      {
        quote &&
          <div className={styles['quote']}>
            {quote}
          </div>
      }

      {
        dropdown &&
          <div
            onClick={dropdown.onClick}
            className={styles['dropdown']}
          >
            <Icon
              icon={dropdown.menuActive ? 'upRebrand' : 'downRebrand'}
              height='8px'
              width='14px'
            />
          </div>
      }
    </div>
  );
}


MobileHeader.propTypes = {
  /**
   * This prop will add a new className to any inherent classNames
   * provided in the component's index.js file.
   */
  className: PropTypes.string,

  /**
   * Main text for header. Centered when alone, left aligned when with other content
   */
  text: PropTypes.node,

  /**
   * Quote information displayed, right aligned
   */
  quote: PropTypes.node,

  /**
   * Allows for dropdown icon in header, right aligned
   */
  dropdown: PropTypes.shape({
    menuActive: PropTypes.bool,
    onClick: PropTypes.func,
  }),
};

MobileHeader.defaultProps = {
  // Place any default props here.
};

export default MobileHeader;
