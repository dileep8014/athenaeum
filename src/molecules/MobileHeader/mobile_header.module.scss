.mobile-header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: ru(2.5);
  background: color('primary-2');
  border-bottom: 1px solid color('neutral-5');
  padding: 0 ru(1);

  .main-text {
    margin: 0 auto;
  }

  .quote {
    display: flex;
    flex-grow: 1;
    justify-content: flex-end;
  }

  .dropdown {
    margin-left: ru(.5);
    cursor: pointer;
  }
}
