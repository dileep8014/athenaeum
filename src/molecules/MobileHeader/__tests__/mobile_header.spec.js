/* eslint-disable no-extra-parens */
import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import MobileHeader from 'molecules/MobileHeader';
import LinkWrapper from 'atoms/LinkWrapper';
import Text from 'atoms/Text';

describe('<MobileHeader />', () => {
  let component;

  beforeEach(() => {
    component =
      (<MobileHeader
        text={
          <LinkWrapper
            type='secondary'
            onClick={jest.fn()}
          >
            <Text
              size={10}
              type='b'
            >
              Save your progress
            </Text>
          </LinkWrapper>
        }
        quote={
          <Text
            size={6}
            font='a'
          >
            <Text tag='sup' size={10} font='a'>$</Text>39.50 <Text tag='span' size={10} font='a'>/mo</Text>
          </Text>
        }
        dropdown={{
          menuActive: false,
          onClick: jest.fn(),
        }}
      />)
    ;
  });

  it('renders correctly', () => {
    const wrapper = mount(component);

    expect(wrapper.type()).toEqual(MobileHeader);
  });

  it('renders correctly', () => {
    const actual = renderer.create(component).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
