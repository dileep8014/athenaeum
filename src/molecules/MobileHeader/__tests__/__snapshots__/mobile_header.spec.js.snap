// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileHeader /> renders correctly 1`] = `
<div
  className="mobile-header"
>
  <div
    className="main-text"
  >
    <a
      className="link neutral-1 undefined secondary"
      onClick={[Function]}
    >
      <p
        className="text typography-10 p undefined type-b-10-bold type-b-10-medium type-b-10-regular"
        onClick={undefined}
      >
        Save your progress
      </p>
    </a>
  </div>
  <div
    className="quote"
  >
    <p
      className="text typography-6 p undefined type-a-6-bold type-a-6-medium type-a-6-regular"
      onClick={undefined}
    >
      <sup
        className="text typography-10 sup undefined type-a-10-bold type-a-10-medium type-a-10-regular"
        onClick={undefined}
      >
        $
      </sup>
      39.50 
      <span
        className="text typography-10 span undefined type-a-10-bold type-a-10-medium type-a-10-regular"
        onClick={undefined}
      >
        /mo
      </span>
    </p>
  </div>
  <div
    className="dropdown"
    onClick={[Function]}
  >
    <span
      className="icon-wrapper"
      onClick={undefined}
      style={
        Object {
          "height": "8px",
          "width": "14px",
        }
      }
    />
  </div>
</div>
`;
